cmd/ci-secret-generator/main.go

```diff
@@ -127,55 +153,104 @@ func executeCommand(command string) ([]byte, error) {
 	return out, nil
 }
 
+func replaceParameter(paramName, param, template string) string {
+	return strings.ReplaceAll(template, fmt.Sprintf("$(%s)", paramName), param)
+}
+
 func updateSecrets(bwItems []bitWardenItem, bwClient bitwarden.Client) error {
-	for _, bwItem := range bwItems {
-		logger := logrus.WithField("item", bwItem.ItemName)
-		if bwItem.Field.Name != "" {
-			logger = logger.WithFields(logrus.Fields{
-				"field":   bwItem.Field.Name,
-				"command": bwItem.Field.Cmd,
-			})
-			logger.Info("processing field")
-			out, err := executeCommand(bwItem.Field.Cmd)
-			if err != nil {
-				logrus.WithError(err).Error("failed to generate field")
-				return err
-			}
-			if err := bwClient.SetFieldOnItem(bwItem.ItemName, bwItem.Field.Name, out); err != nil {
-				logrus.WithError(err).Error("failed to upload field")
-				return err
+	for _, bwItemWithParams := range bwItems {
```

Comment:
Please have a dedicated function that does the resolving and do it before any command is executed, to not mix config resolving steps with program execution steps

Reply:
Furthermore, we want to keep our ability to process everything we can before we exit the program with an error. The way this is built, we will stop processing on the first error here, rather then process the rest